### ISSUE TITLE
TypeError: can only concatenate str (not "NoneType") to str in summarize_worker

### ISSUE THREAD

**Issue by ryanfreckleton** [paul-gauthier/aider#231]
I got the following error when using Aider 0.13.0

```
> Exception in thread Thread-9 (summarize_worker):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()                                                                                                        
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ryan/.local/pipx/venvs/aider-chat/lib/python3.10/site-packages/aider/coders/base_coder.py", line 362, in summarize_worker
    self.summarized_done_messages = self.summarizer.summarize(self.done_messages)
  File "/home/ryan/.local/pipx/venvs/aider-chat/lib/python3.10/site-packages/aider/history.py", line 61, in summarize
    summary = self.summarize_all(head)
  File "/home/ryan/.local/pipx/venvs/aider-chat/lib/python3.10/site-packages/aider/history.py", line 89, in summarize_all
    summary = prompts.summary_prefix + summary
TypeError: can only concatenate str (not "NoneType") to str
```

**Comment by paul-gauthier** [comment]
Thanks for trying aider and reporting this issue.

@joshuavial have you seen this before?

**Comment by joshuavial** [comment]
I saw this error early on in developing the feature but hadn't seen it recently. I did a fair amount of manual testing on the summarizer and it seemed to be working, but I didn't have any unit tests for it. I've just tested again and the history summarizer is working for me off main branch at the moment.

My best guess is that if that the invalidRequestError in simple_send_with_retries is being hit, which would have happened when the openai models were temporarily removed from openrouter last week 

The above PR should handle the error better if it occurs again.

It would be a good idea to get coder summarizing features under test but I don't think that would give us additional insight into why the simple_send_with_retries  returned None in the first place.
